src/rpc/util.cpp

```diff
@@ -86,6 +87,17 @@ CAmount AmountFromValue(const UniValue& value)
     return amount;
 }
 
+CFeeRate FeeRateFromValueInSatB(const UniValue& value)
+{
+    const CAmount amount{AmountFromValue(value)};
+    const CFeeRate fee_rate{CFeeRate::FromSatB(amount)};
+    if (fee_rate.IsZero() && amount != 0) {
+        // passed value is an unrepresentable fee rate between 0 and 0.001 sat/vB
```

Comment:
8885b724eaaba5ce610521d0ee00ce29a16310cc: Whether a feerate is representable or not has nothing to do with 0

9.99999999999999999999999999999 isn't representable and also isn't zero

Reply:
You need to fix this in `AmountFromValue`, as I pointed out earlier: https://github.com/bitcoin/bitcoin/pull/20546#issuecomment-739253473

Reply:
Can you explain how and why this needs to be in `AmountFromValue`? Do we pass in an `is_sat_b_fee_rate` bool to it, so it has context? Is that an improvement over a dedicated function? I feel like I'm failing to read your mind on this since a couple months now.

Reply:
dropped the comment

Reply:
`AmountFromValue` parses the string and fails if the amount is not representable, thus any check about representability should be in `AmountFromValue`.

btc/kb are representable when there are at most 8 digits. sat/b are representable when there are at most 3 digits. How you pass in the digits shouldn't matter. Can be done as normal argument or template argument (and function alias, if this makes the code easier to read).